### PR TITLE
Redshift: drop session schema when test suite finishes

### PR DESCRIPTION
**Pull request by camsaul**
To help with Redshift table exhaustion, have Redshift tests clean up after themselves. Dependent on https://github.com/metabase/hawk/pull/19

I verified this locally and it is working as intended:

```
$ DRIVERS=redshift clj -X:dev:ee:ee-dev:drivers:drivers-dev:user:test :only metabase.driver.redshift-test/correct-driver-test
...
Finding tests took 14.9 s.
Running 1 tests
Running before-run hooks...

0/1     0% [                                                  ]  ETA: --:--Running QP tests against these drivers: #{:redshift}
...
Test took 3.917 seconds seconds to run

1/1   100% [==================================================]  ETA: 00:00

Ran 1 tests in 5.243 seconds
1 assertion, 0 failures, 0 errors.
{:test 1, :pass 1, :fail 0, :error 0, :type :summary, :duration 5242.867726, :parallel 1}
Ran 1 tests in parallel, 0 single-threaded.
Finding and running tests took 20.2 s.
All tests passed.
Running after-run hooks...
[main] INFO  metabase.test.data.interface - Running after-run hooks for :redshift...
[redshift] DROP SCHEMA IF EXISTS "2024_02_07_bb113a43_3dad_40e0_9672_07f0816a47e2_schema" CASCADE;
```

### CODE REVIEW

File: deps.edn [hunk 1]
```diff
@@ -230,7 +230,7 @@
     djblue/portal                {:mvn/version "0.51.0"}                    ; ui for inspecting values
     hashp/hashp                  {:mvn/version "0.2.2"}                     ; debugging/spying utility
     io.github.camsaul/humane-are {:mvn/version "1.0.2"}
-    io.github.metabase/hawk      {:sha "46835a0b808e86894f598472351c08d81aeb1910"}
+    io.github.metabase/hawk      {:local/root "../hawk"}
```
Comment (camsaul) [inline deps.edn:233]: Waiting on https://github.com/metabase/hawk/pull/19 to be approved, then I will update this.

File: modules/drivers/redshift/test/metabase/test/data/redshift.clj [hunk 2]
```diff
@@ -174,6 +174,22 @@
      (delete-old-schemas! conn)
      (create-session-schema! conn))))
 
+(defn- delete-session-schemas!
```
Comment (metamben) [inline modules/drivers/redshift/test/metabase/test/data/redshift.clj:177]:
```suggestion
(defn- delete-session-schema!
```

**Review by metamben — APPROVED** [review]
LGTM

File: deps.edn (hunk 1 above)
Comment (markbastian) [reply deps.edn:233]: ✅ on that pr.

**Review by markbastian — APPROVED** [review]
> Waiting on metabase/hawk#19 to be approved, then I will update this.

Just be sure to update the dep, but otherwise ready to go.

File: modules/drivers/redshift/test/metabase/test/data/redshift.clj (hunk 2 above)
Comment (camsaul) [reply modules/drivers/redshift/test/metabase/test/data/redshift.clj]: done

**Pull request merged**